src/hipscat/catalog/pixel_node.py

```diff
@@ -0,0 +1,86 @@
+from __future__ import annotations
+
+from typing import List
+
+from hipscat.catalog.pixel_node_type import PixelNodeType
+
+
+class PixelNode:
+    """A node in the HIPSCat quadtree of HEALPix pixels
+
+    Attributes:
+        hp_order: HEALPix order of the pixel
+        hp_pixel: HEALPix pixel number
+        node_type: If the node is a leaf, root, or inner type
+        parent: The parent pixel node in the tree
+        children: The child nodes of the pixel
+
+    """
+
+    def __init__(self,
+                 hp_order: int | None,
+                 hp_pixel: int | None,
+                 node_type: PixelNodeType,
+                 parent: PixelNode | None = None,
+                 children: List[PixelNode] = None):
+        """Inits PixelNode with its attributes
+
+        Raises:
+            ValueError: Invalid arguments for the specified pixel type
+        """
+
+        if node_type == PixelNodeType.ROOT and parent is not None:
```

Comment:
Do root nodes have an hp_order we should check? (Alternatively would a given hp_order indicate a root node)?

Reply:
I was thinking using None for the root node, since it's above the order 0 pixels and doesn't really correspond to a HEALPix. I can add a check.

Reply:
With the check mentioned below, order should be -1